### PR TITLE
Streaming buffer restore + and fixes to `EngineStreaming.bufferSize`

### PR DESCRIPTION
**Pull request by Pirulax**
Addendum to #3086
- Now the code correctly restores the buffer size on reconnect.
- Added a function `engineStreamingRestoreBufferSize`
- Fixed `EngineStreming.bufferSize` class var's getter function
- Also refactored `SetStreamingBufferSize` [In an attempt to handle out-of-memory situations correctly, but it turns out the original memory alloc. code just crashes instead]

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 1]
```diff
@@ -128,7 +131,7 @@ void CLuaEngineDefs::LoadFunctions()
         {"engineStreamingRestoreMemorySize", ArgumentParser<EngineStreamingRestoreMemorySize>},
         {"engineStreamingSetBufferSize", ArgumentParser<EngineStreamingSetBufferSize>},
         {"engineStreamingGetBufferSize", ArgumentParser<EngineStreamingGetBufferSize>},
-        {"engineStreamingSetModelCacheLimits", ArgumentParser<EngineStreamingSetModelCacheLimits>},
```
Comment (botder) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:131]: Why was this one removed?

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 2]
```diff
@@ -186,10 +189,10 @@ void CLuaEngineDefs::AddClass(lua_State* luaVM)
         lua_classfunction(luaVM, "restoreMemorySize", "engineStreamingRestoreMemorySize");
         lua_classfunction(luaVM, "getBufferSize", "engineStreamingGetBufferSize");
         lua_classfunction(luaVM, "setBufferSize", "engineStreamingSetBufferSize");
-        lua_classfunction(luaVM, "setModelCacheLimits", "engineStreamingSetModelCacheLimits");
+        lua_classfunction(luaVM, "restoreBufferSize", "engineStreamingSetBufferSize");
```
Comment (botder) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: Why did you remove `setModelCacheLimits`?
Also, shouldn't you use `"engineStreamingRestoreBufferSize"` here?

**Review by botder — CHANGES_REQUESTED** [review]

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp (hunk 2 above)
Comment (Pirulax) [reply Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: Um, perhaps I forgot to pull master.
I'll fix it.

**Review by botder — APPROVED** [review]
